Lib/fontTools/cffLib/__init__.py

```diff
@@ -608,16 +684,46 @@ def produceItem(self, index, data, file, offset):
 		return data
 
 	def append(self, item):
+		"""Add an item to an INDEX."""
 		self.items.append(item)
 
 	def getCompiler(self, strings, parent, isCFF2=None):
 		return self.compilerClass(self, strings, parent, isCFF2=isCFF2)
 
 	def clear(self):
+		"""Empty the INDEX."""
 		del self.items[:]
 
 
 class GlobalSubrsIndex(Index):
+	"""This index contains all the global subroutines in the font. A global
+	subroutine is a set of ``CharString`` data which is accessible to any
+	glyph in the font, and are used to store repeated instructions - for
+	example, components may be encoded as global subroutines, but so could
+	hinting instructions.
+
+	Remember that when interpreting a ``callgsubr`` instruction (or indeed
+	a ``callsubr`` instruction) that you will need to add the "subroutine
+	number bias" to number given:
```

Comment:
Maybe mention that the bias # depends on the number of subrs/gsubrs defined (it's not always 107).